lib/rate-limiter.js

```diff
@@ -0,0 +1,84 @@
+'use strict';
+
+
+/**
+ * module - ratelimiter
+ *
+ * forked from https://github.com/juliendangers/express-limiter2 and updated
+ * to work with Storj framework
+ *
+ * @param  {type} db  redis data client
+ * @param  {type} app express app or express 4.0 router
+ * @return {type}     returns a function that takes config options object
+ */
+module.exports = function(db, app) {
+  return function(opts) {
+    var middleware = function(req, res, next) {
+      if (opts.whitelist && opts.whitelist(req)) {
+        return next();
+      }
+      opts.onRateLimited = typeof opts.onRateLimited === 'function'
+        ? opts.onRateLimited
+        : function(req, res, next) {
+          res.status(429).send('Rate limit exceeded');
+        };
+      // default rate-limit by IP address & method & path
+      opts.lookup = typeof opts.lookup === 'function'
+        ? opts.lookup
+        : function(req) {
+          return [req.path, req.method, req.connection.remoteAddress];
+        };
+      opts.keyFormatter = typeof opts.keyFormatter === 'function'
+        ? opts.keyFormatter
+        : function(parts) {
+          return "rateLimit:" + parts.join(':');
+        };
+      var key = opts.keyFormatter(opts.lookup(req));
+      db.get(key, function(err, limit) {
+        if (err && opts.ignoreErrors) {
+          return next();
```

Comment:
For reference, we have a logger for the error middleware by passing the logger in as an option: https://github.com/Storj/service-middleware/blob/master/lib/error-handler.js#L8